modules/core/src/main/java/org/apache/ignite/internal/marshaller/optimized/OptimizedObjectOutputStream.java

```diff
@@ -180,7 +181,7 @@ private void writeObject0(Object obj) throws IOException {
         if (obj == null)
             writeByte(NULL);
         else {
-            if (obj instanceof Throwable && !(obj instanceof Externalizable)) {
+            if (obj instanceof Throwable && !(obj instanceof Externalizable) || U.isEnum(obj.getClass())) {
```

Comment:
Still think that we should have a comment here on why do we special-case enums. Would be nice to also explain that for throwables, but if we don't know that let's just say "Don't know why Throwable is a special case".